### DISCUSSION TITLE
关于混合检索的测试结果

### DISCUSSION THREAD

**Discussion by Ethereal-sakura** [labring/FastGPT#966] [Q&A]
1. 我自己的数据集有40W条左右，然后放入私有部署的fastgpt的知识库中，当我测试的时候发现，如果采取混合检索，对于结果有的是给出了语义检索和全文检索，有的是只给出了语义检索，请问这个原因是什么呢？
![image](https://github.com/labring/FastGPT/assets/110333012/0441fb96-66de-4ad3-b59b-2ec7f4728c50)
2.通过上面的图片可以明显的看到第二个结果应该是更加符合提问的，所以有什么优化方式吗？目前采取的是文本直接分段的方式存入知识库中。提高向量的维度会更加精确化但是检索效率是否会降低？
3.关于检索效率的提高，目前尝试了对pg数据库的shared_buffers进行增大(变成可用内存的25%)，对于40W的数据(4KW汉字),测试结果明显的从8s到了2.714s，后面会尝试其他参数的影响并分享出来
4.看到别的issue下面作者提到了修改hnsw的参数，通过查看pgvector的仓库，发现可以修改Index Option中的m和ef_construction，但是这个参数都是建表的时候的定义的，请问代码中哪里可以修改？

**Comment by c121914yu** [comment]
> 1. 我自己的数据集有40W条左右，然后放入私有部署的fastgpt的知识库中，当我测试的时候发现，如果采取混合检索，对于结果有的是给出了语义检索和全文检索，有的是只给出了语义检索，请问这个原因是什么呢？
> ![image](https://github.com/labring/FastGPT/assets/110333012/0441fb96-66de-4ad3-b59b-2ec7f4728c50)
> 2.通过上面的图片可以明显的看到第二个结果应该是更加符合提问的，所以有什么优化方式吗？目前采取的是文本直接分段的方式存入知识库中。提高向量的维度会更加精确化但是检索效率是否会降低？
> 3.关于检索效率的提高，目前尝试了对pg数据库的shared_buffers进行增大(变成可用内存的25%)，对于40W的数据(4KW汉字),测试结果明显的从8s到了2.714s，后面会尝试其他参数的影响并分享出来
> 4.看到别的issue下面作者提到了修改hnsw的参数，通过查看pgvector的仓库，发现可以修改Index Option中的m和ef_construction，但是这个参数都是建表的时候的定义的，请问代码中哪里可以修改？

1  全文检索是有空的结果，不是每条数据都会有结果。
2 增加自定义索引
4 索引参数固定的。如果需要，可以连接数据库删除再重建。

**Reply by Ethereal-sakura** [reply]
您好，非常感谢您的回复，关于您的回复，我有几点不懂
1.自定义索引能通过前端添加吗，我看文档上只有通过API的方式提到了自定义索引
2.目前所有的知识库都是在一张表上面的吗，这样删除表就相当于删除了所有知识库，是否会考虑一个知识库对应一张表呢？
